casa/OS/DynLib.cc

```diff
@@ -152,8 +124,91 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     }
   }
 
+  std::string DynLib::tryOpen (const std::string& library,
+                               const std::string& dir,
+                               const std::string& prefix,
+                               const std::string& version)
+  {
+    std::string pref(prefix);
+    std::string vers(version);
+    std::string fullName;
+    // Try a maximum 4 times (1 or 2 prefix, 1 or 2 version, 1 ext).
+    int i=0;
+    while (i<4  &&  itsHandle==0) {
+#ifdef __APPLE__
+      fullName = dir + pref + library + vers + ".dylib";
+#else
+      fullName = dir + pref + library + ".so" + vers;
+#endif
+      open (fullName);
+      i++;
+      if (i == 2) {
+        if (pref == "lib") i += 2;     // no specific prefix given
+        pref = "lib";
+      }
+      if (i%2 == 1) {
+        vers = std::string();
+        if (version.empty()) i++;      // no version given
+      } else {
+        vers = version; 
+      }
+    }
+    return (itsHandle==0  ?  std::string() : fullName);
+  }
+
+  void DynLib::attach (const std::string& library,
+                       const std::string& prefix,
+                       const std::string& version,
+                       const std::string& funcName)
+  {
+    std::string fullName = tryCasacorePath (library, prefix, version);
+    if (fullName.empty()) {
+      fullName = tryOpen (library, string(), prefix, version);
+    }
+    if (itsHandle == 0) {
+      throw AipsError ("Shared library " + library +
+                       " found in CASACORE_LDPATH nor (DY)LD_LIBRARY_PATH\n"
```

Comment:
should probably be `not found?
Also the error should state the version and prefix that has not been found